crates/polars-core/src/utils/mod.rs

```diff
@@ -34,6 +35,35 @@ impl<T> Deref for Wrap<T> {
     }
 }
 
+// This is giant trust me bro
+#[repr(transparent)]
+#[derive(Debug)]
+pub struct RegexWrap<T>(pub T);
```

Comment:
I recommend having the original pattern as well here, and calling the compare function/hasing it